include/ginkgo/core/preconditioner/batch_ilu.hpp

```diff
@@ -164,21 +170,34 @@ class BatchIlu : public EnableBatchLinOp<BatchIlu<ValueType, IndexType>>,
      * @param system_matrix  the source matrix used to generate the
      *                       preconditioner
      */
-    void generate(const BatchLinOp* system_matrix);
+    void generate_precond(const BatchLinOp* system_matrix);
+
+    // Since there is no guarantee that the complete generation of the
+    // preconditioner would occur outside the solver kernel, that is in the
+    // external generate step, there is no logic in implementing "apply" for
+    // batched preconditioners
+    void apply_impl(const BatchLinOp* b, BatchLinOp* x) const override
+        GKO_BATCHED_NOT_SUPPORTED(
+            "batched preconditioners do not support (user facing/public) "
+            "apply");
 
-    void apply_impl(const BatchLinOp* b, BatchLinOp* x) const override{};
 
     void apply_impl(const BatchLinOp* alpha, const BatchLinOp* b,
-                    const BatchLinOp* beta, BatchLinOp* x) const override{};
+                    const BatchLinOp* beta, BatchLinOp* x) const override
```

Comment:
This is something I would note in the class documentation as well.

Reply:
Added to the class docs.